package.json

```diff
@@ -17,6 +17,7 @@
     "async": "0.2.6",
     "atom-keymap": "6.3.2",
     "babel-core": "^5.8.21",
+    "binary-search-with-index": "^1.3.0",
```

Comment:
I'd prefer not to pull in a dependency for binary search, especially since that package contains extra stuff we don't need in this code path (such as checking for `undefined`). I'd prefer having something like:

```js
function binarySearch (array, compare) {
  let low = 0
  let high = array.length
  while (low <= high) {
    let mid = low + (high - low >> 1)
    let comparison = compare(array[mid], mid)
    if (comparison < 0) {
      low = mid + 1
    } else if (comparison === 0) {
      return mid
    } else {
      high = mid - 1
    }
  }

  return -1
}
```

It's not exactly DRY, but the cost of depending on a library seems higher in this case.